.github/workflows/test.yml

```diff
@@ -4,29 +4,103 @@ on: [push, pull_request]
 
 env:
   CARGO_TERM_COLOR: always
+  # CI builds are closer to a from-scratch build anyways
+  # and incremental compilation hurts reproducibility
+  CARGO_INCREMENTAL: 0
 
 jobs:
-  test:
+  unit-tests:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest]
     steps:
     - uses: actions/checkout@v2
-    - name: Unit tests
+
+    - uses: Swatinem/rust-cache@v1.3.0
+
+    - name: Build kak-lsp
+      run: cargo test --no-run --locked --release
+
+    - name: Run unit tests
       run: cargo test --release
-    - name: Install system test dependencies
+
```

Comment:
I would call them system tests because they test exactly what the user runs - an actual instance of Kakoune with real language servers.
Though this really doesn't matter, no need to change it.

In future we could have integration tests that test the kak-lsp with mock LSP messages.